### ISSUE TITLE
[Android Project]: Assignment 2: Newyork times article seach - ready for review

### ISSUE THREAD

**Issue by nigamg** [nigamg/NYArticleSearchClient#1]
My app is ready, please review ./cc @codepathreview @codepath"

**Comment by codepathreview** [comment]
@nigamg The requirement "User can scroll down "infinitely" to continue loading more news articles" is not implemented. Pls update the status here

/cc @nidhi1608

**Comment by nigamg** [comment]
Updated the walkthrough and readme covering more cases. /cc @nidhi1608

**Comment by codepathreview** [comment]
@nigamg The user story mentioned above is still not implemented.

/cc @nidhi1608

**Comment by nigamg** [comment]
@nidhi1608, Implemented infinite scrolling but it gets called initial load but it is not getting called in any subsequent scroll down. Actually onScroll event is not getting called even through listener is attached successfully. Debugging.

**Comment by nesquena** [comment]
Can you check the hints for this week about scrolling there's a bunch of debugging tips. In particular make sure you are clearing the array at only at the appropriate time as outlined in the hints. In your current code I don't see you clearing the array on fresh searches. You want to make sure you aren't clearing on pagination and instead appending results to existing items.

**Comment by nesquena** [comment]
Also avoid resetting the adapter on every pagination to avoid "setAdapter" on subsequent searches

**Comment by rogerhu** [comment]
Yeah the creation of a new adapter each time is likely to be the issue.

**Comment by nigamg** [comment]
thank you for a suggestions, I tried to remove setAdapter from subsequent searches. But then I am not getting search results displayed in the grid. 
Still debugging.

**Comment by rogerhu** [comment]
notifyItemInsertedRange should also have as the 2nd parameter article size, not article size - 1

**Comment by nigamg** [comment]
finally I figured out the problem, I was creating new StaggeredLayoutManager in search result and I guess, because of layoutmanager, scrollView listener was not getting attached. 

problem fixed but now items are getting twice in very first initialization, should be a small glitch, debugging.

**Comment by nigamg** [comment]
@rogerhu  @nesquena  thank you all for all the steps suggested.

**Comment by nigamg** [comment]
I am sorry, it seems, it is still not working, onScroll listener is not getting called  :( 
It worked last time, when I got the so many results covering the screen, rather grid items were at the bottom (aligned to bottom) and onScroll listener got invoked.

**Comment by nesquena** [comment]
You are clearing in the wrong place: https://github.com/nigamg/NYArticleSearchClient/blob/master/app/src/main/java/com/gn/demo/nyarticlesearchclient/activity/NYArticleSearchActivity.java#L120-L121

Don't clear on pagination, only on a fresh search....
